### DISCUSSION TITLE
Retrieving Transformation Matrix Used to Stitch Mosaic?

### DISCUSSION THREAD

**Discussion by ericvaughn11** [OpenStitching/stitching#161] [Q&A]
Hi,

Thanks for the excellent implementation and tutorial! I've successfully used the package to stitch an 4x4 array of images using `AffineStitcher`, but I also need the pixels used for each of the 16 images for filtering data that I will overlay onto this mosaic. Is there a way to easily retrieve the transformation matrix used to stitch the 16 images?

Thanks,
Eric

**Comment by lukasalexanderweber** [comment]
Hey Eric,

I'm quite sure you'll get the transformation matrix from the `cameras`.

See e.g. [here](https://github.com/whdlgp/Stitching-tutorial-with-OpenCV/blob/master/get_camera_params/get_camera_params.md)

**Comment by ericvaughn11** [comment]
Thanks for the prompt response. I was able to dig into this a little more with `stitch_verbose`. I tried looking through cameras documentation, but I'm a bit lost on the C++. I'm trying to implement something that would be an automated script within Python so keeping everything within Python would be greatly preferred (especially since I don't know C++). 

Going through `stitch_verbose` and the jupyter notebook, I'm wondering if there's a way to get images that retain their original dimensionality, but only see `create_and_warp_masks` used, which changes the image sizes.  `final_corners` and an originally sized (2720x2720) images would be what I'd need. I'm filtering other data from each tile to overlay onto the final mosaic, so any change in x/y makes it a bit more difficult to map back to other 2720x2720 arrays for each FOV.
